### ISSUE TITLE
Patch sequence that resets array then adds to it results in incorrect update result 

### ISSUE THREAD

**Issue by chrisrichard** [mrcranky/rfc6902-mongodb#2]
I think we found another edge case that's a bit off:

```
import {updatesForPatch} from "https://esm.sh/rfc6902-mongodb";

const result = updatesForPatch([
        { op: 'replace', path: '/addresses', value: []},
        { op: 'add', path: '/addresses/-', value: '1'},
        { op: 'add', path: '/addresses/-', value: '2'}
    ], { addresses: ['a', 'b']})
console.log(JSON.stringify(result));
```

Actual:
`>>> [{"$set": {"addresses": ["1", "2"]}}, {"$push": {"addresses": {"$each": ["1", "2"]}}}] `

Expected:
`>>> [{"$set": {"addresses": []}}, {"$push": {"addresses": {"$each": ["1", "2"]}}}] `
or
`>>> [{"$set": {"addresses": ["1", "2"]}}]`

**Comment by mrcranky** [comment]
Hmm, interesting, this looks like it's a weird behaviour from the underlying rfc6902 module.
```
const patch = [
    { 'op': 'replace', 'path': '/foo', 'value': [] },
    { 'op': 'add', 'path': '/foo/-', 'value': 'bar' },
];
console.log(patch[0].value);
applyPatch({}, patch);
console.log(patch[0].value);
applyPatch({foo: []}, patch);
console.log(patch[0].value);
```
produces
```
(0) []
(0) []
(1) ['bar']
```
If you use replace to set a field to an array value, then use add to modify that array, it will modify the patch document that has been passed in, **but only if the field already existed in the document being changed**. That feels like it is probably a bug in that module, or at least unexpected / undocumented behaviour.

**Comment by mrcranky** [comment]
Added a clone of the patch document just prior to use which works around that defect for now, will raise an issue over in that project and hopefully if it gets tackled I can back out the inefficient copying

**Comment by mrcranky** [comment]
Raised https://github.com/chbrown/rfc6902/issues/97 and a PR over there, if that gets accepted then I'll pick it up and make a new version here.

**Comment by mrcranky** [comment]
Actually @chrisrichard, I was waiting till folks were using this in earnest before declaring 1.0.0; as far as I'm concerned it's feature complete. Aside from the couple of issues you've raised, is it working well enough for you that you'd say it's fit for purpose? If so I'll cut a 1.0.0 release so integrators don't feel like it's pre-release.

**Comment by chrisrichard** [comment]
Yes, working quite well for us. 

The only other issue for us is we do still need the commonjs module format for the time being. Someone has published a rfc6902-mongodb-commonjs package (https://www.npmjs.com/package/rfc6902-mongodb-commonjs) but would be nice to include this in the published 1.0.0 package if possible.

**Comment by mrcranky** [comment]
I'd hoped to avoid a build step and keep it all clean and simple, but I appreciate the stickiness of CommonJS projects. 0.2.0 is double the size (CJS + ESM forms) but should remove the need for the wrapped package. 1.0.0 will be cut next week barring any issues found in 0.2.0
